doc/porting/board_porting.rst

```diff
@@ -229,12 +234,8 @@ To do this, we extend the definition of ``FOO_WIDTH`` as follows, in
 ``default`` properties from :file:`Kconfig.defconfig` files override
 ``default`` properties given on the "base" definition of the symbol. Zephyr
 uses a custom Kconfig patch that makes Kconfig prefer later defaults, and
-includes the :file:`Kconfig.defconfig` files last.
-
-.. note::
-
-    Assignments in :file:`.config` files have no effect on invisible symbols,
-    so this scheme is not just an organizational issue.
+includes the :file:`Kconfig.defconfig` files last. See the
```

Comment:
should this be ``file`` and not plural?  also three lines up?  Can there be more than one ``Kconfig.defconfig`` file?

Reply:
Usually, there's a single configuration tree with all the Kconfig files in it, as you can't do a "conditional" include in Kconfig (would get messy, as you'd need to evaluate stuff before you have even seen all the symbols). Dependencies take care of making "irrelevant" stuff disappear.

Zephyr prunes the configuration tree before parsing though, by setting some environment variables that are then expanded in `source` statements. Saves a bit of time.

I'm fine with changing it to "file". Might be less confusing.

Reply:
your call.  I see why "files" would be OK too.